web/client-api/src/main/java/io/deephaven/web/client/api/parse/JsDataHandler.java

```diff
@@ -0,0 +1,488 @@
+                    nullCount++;
+                } else {
+                    nulls.set(i);
+                }
+                payload.setAt(i, Double.longBitsToDouble(dateValue));
+            }
+
+            // validity, then payload
+            addBuffer.apply(makeValidityBuffer(nullCount, nulls));
+            addBuffer.apply(new Uint8Array(payload.buffer));
+
+            addNode.apply(new Node(data.length, nullCount));
+        }
+    },
+    INTEGER(Type.Int, "int") {
+        @Override
+        public double writeType(Builder builder) {
+            return Int.createInt(builder, 32, true);
+        }
+
+        @Override
+        public void write(Object[] data, ParseContext context, JsConsumer<Node> addNode,
+                JsConsumer<Uint8Array> addBuffer) {
+            writeSimpleNumbers(data, addNode, addBuffer, Int32Array.BYTES_PER_ELEMENT, NULL_INT, Int32Array::new);
+        }
+    },
+    SHORT(Type.Int, "short") {
+        @Override
+        public double writeType(Builder builder) {
+            return Int.createInt(builder, 16, true);
+        }
+
+        @Override
+        public void write(Object[] data, ParseContext context, JsConsumer<Node> addNode,
+                JsConsumer<Uint8Array> addBuffer) {
+            writeSimpleNumbers(data, addNode, addBuffer, Int16Array.BYTES_PER_ELEMENT, NULL_SHORT, Int16Array::new);
+        }
+    },
+    LONG(Type.Int, "long") {
+        @Override
+        public double writeType(Builder builder) {
+            return Int.createInt(builder, 64, true);
+        }
+
+        @Override
+        public void write(Object[] data, ParseContext context, JsConsumer<Node> addNode,
+                JsConsumer<Uint8Array> addBuffer) {
+            int nullCount = 0;
+            BitSet nulls = new BitSet(data.length);
+            // using float because we can convert longs to
+            // doubles, though not cheaply
+            Float64Array payload = new Float64Array(data.length);
+            for (int i = 0; i < data.length; i++) {
+                final long value;
+                if (data[i] == null) {
+                    value = NULL_LONG;
+                } else if (data[i] instanceof LongWrapper) {
+                    value = ((LongWrapper) data[i]).getWrapped();
+                } else if (Js.typeof(data[i]).equals("string")) {
+                    String str = (String) data[i];
+                    if (str.trim().length() == 0) {
+                        value = NULL_LONG;
+                    } else {
+                        value = Long.parseLong(str);
+                    }
+                } else {
+                    // fall back to whatever we can get - this is null safe, since we already tested for null above
+                    value = (long) (double) JsDataHandler.doubleFromData(data[i]);
+                }
+                if (value == NULL_LONG) {
+                    nullCount++;
+                } else {
+                    nulls.set(i);
+                }
```

Comment:
It's ok actually to count all of these as not null (for the count). If the null count is zero, then the validity buffer can be skipped (sent as empty buffer). Server side, whether the client enables sending deephaven nulls or not, these longs will simply be the null value. So to save bandwidth, I recommend we do this.